server/schedule/filter/candidates.go

```diff
@@ -39,12 +40,24 @@ func (c *StoreCandidates) FilterSource(opt *config.PersistOptions, filters ...Fi
 	return c
 }
 
+// FilterSourceWithCollector keeps stores that can pass all source filters.
+func (c *StoreCandidates) FilterSourceWithCollector(opt *config.PersistOptions, collector *plan.Collector, filters ...Filter) *StoreCandidates {
+	c.Stores = SelectSourceStoresWithCollector(c.Stores, filters, opt, collector)
+	return c
+}
+
 // FilterTarget keeps stores that can pass all target filters.
 func (c *StoreCandidates) FilterTarget(opt *config.PersistOptions, filters ...Filter) *StoreCandidates {
 	c.Stores = SelectTargetStores(c.Stores, filters, opt)
 	return c
 }
 
+// FilterTargetWithCollector keeps stores that can pass all target filters.
+func (c *StoreCandidates) FilterTargetWithCollector(opt *config.PersistOptions, collector *plan.Collector, filters ...Filter) *StoreCandidates {
```

Comment:
I perfer to merge `FilterTargetWithCollector` with `FilterTarget` with the following reasons:
1. make the diagnose logic the same with product-logic 
2. we do not need to maintain the other function once we need to change any one of them.